File: src-cpp/app/modules/startup/onboarding/item.cpp
#include "item.h"
#include <QString>

namespace Modules
{
namespace Startup
{
namespace Onboarding
{
Item::Item(QString id, QString alias, QString identicon, QString address, QString keyUid)
	: m_id(id)
	, m_alias(alias)
	, m_identicon(identicon)
	, m_address(address)
	, m_keyUid(keyUid)
{ }
QString Item::getId()
{
	return m_id;
}
QString Item::getAlias()
{
	return m_alias;
}
QString Item::getIdenticon()
{
	return m_identicon;
}
QString Item::getAddress()
{
	return m_address;
}
QString Item::getKeyUid()
{
	return m_keyUid;
}
} // namespace Onboarding
} // namespace Startup
} // namespace Modules

File: src/app/modules/startup/onboarding/item.nim
type
  Item* = object
    id: string
    alias: string
    address: string
    pubKey: string
    keyUid: string

proc initItem*(id, alias, address, pubKey, keyUid: string): Item =
  result.id = id
  result.alias = alias
  result.address = address
  result.pubKey = pubKey
  result.keyUid = keyUid

proc getId*(self: Item): string =
  return self.id

proc getAlias*(self: Item): string =
  return self.alias

proc getAddress*(self: Item): string =
  return self.address

proc getPubKey*(self: Item): string =
  return self.pubKey

proc getKeyUid*(self: Item): string =
  return self.keyUid
